### ISSUE TITLE
incorrect error message

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#415]
This error message is incorrect: https://github.com/akuity/kargo/blob/main/internal/controller/promotions/webhook.go#L78

It reads:

```
error indexing Secrets by repo
```

It should be:

```
error indexing PromotionPolicies by Environment
```

**Comment by karthickbharathi1** [comment]
Hi @krancour can you assign this issue to me?

**Comment by krancour** [comment]
This was closed by #421
